activerecord/test/cases/associations/belongs_to_associations_test.rb

```diff
@@ -422,6 +422,24 @@ def test_belongs_to_with_touch_option_on_touch
     assert_queries(1) { line_item.touch }
   end
 
+  def test_belongs_to_with_touch_on_multiple_records
+    line_item = LineItem.create!(amount: 1)
+    line_item2 = LineItem.create!(amount: 2)
+    Invoice.create!(line_items: [line_item, line_item2])
+
+    assert_queries(1) do
+      LineItem.transaction do
+        line_item.touch
+        line_item2.touch
+      end
+    end
+
+    assert_queries(2) do
+      line_item.touch
+      line_item2.touch
+    end
+  end
+
```

Comment:
Need tests demonstrating that
1. a non-persisted record isn't touched before commit
2. the touched timestamp is the expected `current_time_from_proper_timezone`
3. both named attributes and `timestamp_attributes_for_update_in_model` are touched
4. calling `#touch_later` on a non-persisted record raises (arguably we could relax this, since the record may be persisted by the time the txn commits!)
5. that `#touch_later` on an associated record doesn't cause it to get autosaved by the main record save
6. that `#touch_later` doesn't mark the record as changed, or the timestamp attributes as changed
7. that calling `#touch_now(*names)` touches both `names` as well as any outstanding deferred touches
8. that calling `#touch_later` doesn't immediately hit the db, that it's triggered before commit